WMF Framework/Remote Notifications/Model/RemoteNotificationsModelController.swift

```diff
@@ -102,22 +167,27 @@ final class RemoteNotificationsModelController: NSObject {
         backgroundContext.mergePolicy = NSMergeByPropertyObjectTrumpMergePolicy
         return backgroundContext
     }
-
-    private func save(moc: NSManagedObjectContext) {
-        if moc.hasChanges {
-            do {
-                try moc.save()
-            } catch let error {
-                DDLogError("Error saving RemoteNotificationsModelController managedObjectContext: \(error)")
-            }
-        }
+    
+    var numberOfUnreadNotifications: Int? {
+        let fetchRequest: NSFetchRequest<RemoteNotification> = RemoteNotification.fetchRequest()
+        fetchRequest.predicate = unreadNotificationsPredicate
+        return try? viewContext.count(for: fetchRequest)
     }
 
     public func createNewNotifications(moc: NSManagedObjectContext, notificationsFetchedFromTheServer: Set<RemoteNotificationsAPIController.NotificationsResult.Notification>, completion: @escaping () -> Void) throws {
         moc.perform {
             for notification in notificationsFetchedFromTheServer {
                 self.createNewNotification(moc: moc, notification: notification)
             }
+            
+            let newPageContainsUnread = notificationsFetchedFromTheServer.contains(where: { notification in
+                notification.readString != nil
+            })
+                
+            if newPageContainsUnread {
+                NotificationCenter.default.post(name: Notification.Name.NotificationsCenterBadgeNeedsUpdate, object: nil)
+            }
+            
             self.save(moc: moc)
             completion()
         }
```

Comment:
@tonisevener I was noticing some strange behavior around the in-app explore feed bell not properly updating to its indicated state (bell with dot indicator) when a user initially logs in, has the initial import happen, and has unread notifications. The explore feed properly updates to show the bell, but it's in the wrong (un-indicated) state. 

I think this suggestion can address that issue, but I wanted to check with you before I commit – do you see any drawbacks with it I'm missing?

```suggestion
    public func createNewNotifications(moc: NSManagedObjectContext, notificationsFetchedFromTheServer: Set<RemoteNotificationsAPIController.NotificationsResult.Notification>, completion: @escaping () -> Void) throws {
        moc.perform {
            for notification in notificationsFetchedFromTheServer {
                self.createNewNotification(moc: moc, notification: notification)
            }
            
            self.save(moc: moc)
            NotificationCenter.default.post(name: Notification.Name.NotificationsCenterBadgeNeedsUpdate, object: nil)
            completion()
        }
    }
```

Reply:
@staykids ah, good catch! Your suggestion looks good. I think I was just trying to optimize and limit the `NSNotification` from firing needlessly (and thus an unnecessary count fetch from the database and view update), but now that I think about it, we do want it to update whether or not it contains an unread notification. I could see a situation where it's in a badged state, but a refresh call updates all notifications in the database to be read (if say they marked them from Desktop). We'd need this to fire and un-badge the bell icon.